api/src/lib.rs

```diff
@@ -245,6 +248,45 @@ impl InternedStringId {
     }
 }
 
+/// A mechanism for caching interned string IDs.
+pub struct CachedInternedStringId {
+    value: &'static str,
+    interned_string_id: AtomicUsize,
+    context: AtomicPtr<std::ffi::c_void>,
+}
+
+impl CachedInternedStringId {
+    /// Create a new cached interned string ID.
+    pub const fn new(value: &'static str) -> Self {
+        Self {
+            value,
+            interned_string_id: AtomicUsize::new(0),
```

Comment:
If we're keeping this a usize, could we choose a better default value? At construction time we don't know the id, but 0 might as well be a valid id, right? Perhaps we could use `usize::MAX`?